### PR TITLE
Resolves #319, API: Tree Processor extension

### LINKED ISSUE #319: Map the API needed to write a tree processor extension
Map the APIs needed to write a tree processor extension.

It should be possible to write the following (pseudo-code):

```javascript
Opal.Asciidoctor.$$scope.Extensions.register(function() {
  var self = this;
  this.treeProcessor(function() {
    self.process(function(doc) {
      return doc;
    });
  });
});
```

A treeprocessor can return a replacement document or null to continue using the same document.

### PR DESCRIPTION
**Pull request by Mogztter**

### CODE REVIEW

File: spec/share/extensions/love-tree-processor.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = function (registry) {
```
Comment (Mogztter) [inline spec/share/extensions/love-tree-processor.js]: @hsablonniere this is for you :heart:

File: src/asciidoctor-extensions-api.js [hunk 2]
```diff
@@ -45,14 +48,19 @@ var toBlock = function (block) {
...
+// Alias
+Opal.Asciidoctor.Extensions = Extensions;
+
 /**
  * @memberof Extensions
  */
 Extensions.create = function (name, block) {
   if (typeof name === 'function' && typeof block === 'undefined') {
-    return Opal.send(this, 'build_registry', null, toBlock(name));
+    return Opal.send(this, 'create', null, toBlock(name));
```
Comment (Mogztter) [inline src/asciidoctor-extensions-api.js]: @mojavelinux `build_registry` is deprecated so now I'm using `create` and I've also added a case when the block is undefined.

Comment (mojavelinux) [inline src/asciidoctor-extensions-api.js]: The reason I mapped to build_registry was so that we can still build against 1.5.5. Are we still doing that?

Comment (Mogztter) [inline src/asciidoctor-extensions-api.js]: Yes we are! Travis will punch me in the face :fist_oncoming:

**Pull request merged**